charger/nrggen2.go

```diff
@@ -0,0 +1,365 @@
+			nrg.maxCurrentMillis(current)
+		} else {
+			nrg.MaxCurrent(int64(current))
+		}
+	}
+
+	if b, err := nrg.conn.ReadHoldingRegisters(nrgKickGen2MaxPhases, 1); err == nil {
+		if maxPhases := encoding.Uint16(b); maxPhases > 1 {
+			if currentPhases, err := nrg.GetPhases(); err == nil {
+				if err := nrg.phases1p3p(2); err == nil {
+					phasesS = nrg.phases1p3p
+				}
+
+				if phasesS != nil {
+					nrg.phases1p3p(currentPhases)
+				}
+			}
+		}
+	}
+
+	nrg.Enable(enabled)
+
+	return decorateNRGKickGen2(nrg, chargerEx, phasesS), nil
+}
+
+// NewNRGKickGen2 creates NRGKickGen2 charger
+func NewNRGKickGen2(uri string, slaveID uint8) (*NRGKickGen2, error) {
+	conn, err := modbus.NewConnection(uri, "", "", 0, modbus.Tcp, slaveID)
+	if err != nil {
+		return nil, err
+	}
+
+	if !sponsor.IsAuthorized() {
+		return nil, api.ErrSponsorRequired
+	}
+
+	log := util.NewLogger("nrggen2")
+	conn.Logger(log.TRACE)
+
+	nrg := &NRGKickGen2{
+		conn: conn,
+	}
+
+	return nrg, nil
+}
+
+// Status implements the api.Charger interface
+func (nrg *NRGKickGen2) Status() (api.ChargeStatus, error) {
+	b, err := nrg.conn.ReadHoldingRegisters(nrgKickGen2RegStatus, 1)
+	if err != nil {
+		return api.StatusNone, err
+	}
+
+	// 0 - "UNKNOWN",
+	// 1 - "STANDBY",
+	// 2 - "CONNECTED",
+	// 3 - "CHARGING",
+	// 6 - "ERROR",
+	// 7 - "WAKEUP"
+	switch status := encoding.Uint16(b); status {
+	case 0:
+		return api.StatusNone, nil
+	case 1:
+		return api.StatusA, nil
+	case 2:
+		return api.StatusB, nil
+	case 3:
+		return api.StatusC, nil
+	case 6:
+		// 0 - "NO_ERROR",
+		// 1 - "GENERAL_ERROR",
+		// 2 - "32A_ATTACHMENT_ON_16A_UNIT",
+		// 3 - "VOLTAGE_DROP_DETECTED",
+		// 4 - "UNPLUG_DETECTION_TRIGGERED",
+		// 5 - "TYPE2_NOT_AUTHORIZED",
+		// 16 - "RESIDUAL_CURRENT_DETECTED",
+		// 32 - "CP_SIGNAL_VOLTAGE_ERROR",
+		// 33 - "CP_SIGNAL_IMPERMISSIBLE",
+		// 34 - "EV_DIODE_FAULT",
+		// 48 - "PE_SELF_TEST_FAILED",
+		// 49 - "RCD_SELF_TEST_FAILED",
+		// 50 - "RELAY_SELF_TEST_FAILED",
+		// 51 - "PE_AND_RCD_SELF_TEST_FAILED",
+		// 52 - "PE_AND_RELAY_SELF_TEST_FAILED",
+		// 53 - "RCD_AND_RELAY_SELF_TEST_FAILED",
+		// 54 - "PE_AND_RCD_AND_RELAY_SELF_TEST_FAILED",
+		// 64 - "SUPPLY_VOLTAGE_ERROR",
+		// 65 - "PHASE_SHIFT_ERROR",
+		// 66 - "OVERVOLTAGE_DETECTED",
+		// 67 - "UNDERVOLTAGE_DETECTED",
+		// 68 - "OVERVOLTAGE_WITHOUT_PE_DETECTED",
+		// 69 - "UNDERVOLTAGE_WITHOUT_PE_DETECTED",
+		// 70 - "UNDERFREQUENCY_DETECTED",
+		// 71 - "OVERFREQUENCY_DETECTED",
+		// 72 - "UNKNOWN_FREQUENCY_TYPE",
+		// 73 - "UNKNOWN_GRID_TYPE",
+		// 80 - "GENERAL_OVERTEMPERATURE",
+		// 81 - "HOUSING_OVERTEMPERATURE",
+		// 82 - "ATTACHMENT_OVERTEMPERATURE",
+		// 83 - "DOMESTIC_PLUG_OVERTEMPERATURE",
+		// x - "UNKNOWN"
+		b, err := nrg.conn.ReadHoldingRegisters(nrgKickGen2RegError, 1)
+		if err != nil {
+			return api.StatusNone, err
+		}
+
+		switch error := encoding.Uint16(b); error {
+		case 32:
+			fallthrough
+		case 33:
+			return api.StatusE, fmt.Errorf("%d", error)
```

Comment:
I'd omit StatusE since it isn't handled by evcc- use F instead.

Reply:
i'll change, but there are some charges which are using the StatusE. Just want to point out

Reply:
That's true- and thankfully these don't appear in practice ;)